R/zchunk_LB111.ag_resbio_R_C.R

```diff
@@ -32,39 +33,47 @@ module_aglu_LB111.ag_resbio_R_C_DISABLED <- function(command, ...) {
 
-    # ===================================================
-    # TRANSLATED PROCESSING CODE GOES HERE...
-    #
-    # If you find a mistake/thing to update in the old code and
-    # fixing it will change the output data, causing the tests to fail,
-    # (i) open an issue on GitHub, (ii) consult with colleagues, and
-    # then (iii) code a fix:
-    #
-    # if(OLD_DATA_SYSTEM_BEHAVIOR) {
-    #   ... code that replicates old, incorrect behavior
-    # } else {
-    #   ... new code with a fix
-    # }
-    #
-    #
-    # NOTE: there are `merge` and/or 'match' calls in this code. Be careful!
-    # For more information, see https://github.com/JGCRI/gcamdata/wiki/Merge-and-Match
-    # ===================================================
+    # Compute weighted averages of each parameter ( HarvestIndex, ErosionControl, and ResidueEnergyContent ) for each crop type in each GCAM region
+    L100.FAO_ag_Prod_t %>%
+      select(iso, item, year, value) %>%
+      filter(year %in% FINAL_HISTORICAL_YEAR) %>%
+      select(-year) %>%
+      rename(prod = value) %>%
+      full_join(Various_ag_resbio_data, by = "item") %>%
+    # Drop rows with NA values ( dropping commodities that are not in the resbio dataset )
+      na.omit() %>%
+    # also drop rows where production weights are zero, as these would return missing values later on
+      filter(prod != 0) %>%
+    # Multiply by production to get weights, change to long-format for easier calculation
+      gather(resbio_params, value, -c(iso, item, prod)) %>%
+      mutate(value = value * prod) %>%
+    # Add vectors for GCAM regions and commodities, collapse, and divide by production to get residue biomass values
+      left_join(iso_GCAM_regID[c( "iso", "GCAM_region_ID")], by = "iso") %>%
+      left_join(FAO_ag_items_PRODSTAT[c( "item", "GCAM_commodity")], by = "item") %>%
+      group_by(GCAM_region_ID, GCAM_commodity, resbio_params) %>%
+      summarize_if(is.numeric, sum) %>%
+    # Dividing by production to get weighted average residue biomass parameters by region and crop
+      ungroup() %>%
+      mutate(value = value / prod) %>%
+      select(-prod) %>%
+      spread(resbio_params, value) %>%
 
     # Produce outputs
     # Temporary code below sends back empty data frames marked "don't test"
     # Note that all precursor names (in `add_precursor`) must be in this chunk's inputs
     # There's also a `same_precursors_as(x)` you can use
     # If no precursors (very rare) don't call `add_precursor` at all
-    tibble() %>%
-      add_title("descriptive title of data") %>%
-      add_units("units") %>%
-      add_comments("comments describing how data generated") %>%
-      add_comments("can be multiple lines") %>%
+      add_title("Weighted average residue biomass parameters by GCAM region / commodity") %>%
+      add_units("units: varied") %>%
+      add_comments("Weighted average residue biomass parameters by GCAM region / commodity") %>%
```

Comment:
In addition, can you clarify in the comments what the residue biomass parameters are?